### PR TITLE
[Done] Translate "Reconciliation"

### PR DESCRIPTION
**Pull request by Fcmam5**
~~:construction:  Work in progress ..~~ Done

### CODE REVIEW

File: content/docs/reconciliation.md [hunk 1]
```diff
@@ -1,27 +1,27 @@
 ---
 id: reconciliation
-title: Reconciliation
+title: المطابقة (Reconciliation)
 permalink: docs/reconciliation.html
 ---
 
-React provides a declarative API so that you don't have to worry about exactly what changes on every update. This makes writing applications a lot easier, but it might not be obvious how this is implemented within React. This article explains the choices we made in React's "diffing" algorithm so that component updates are predictable while being fast enough for high-performance apps.
+تُزوّدنا React بواجهة برمجة تطبيقات (API) صريحة بحيث لا نقلق بشأن التغييرات التي تطرأ في كل تحديث. يجعل هذا من كتابة التطبيقات أمرًا أسهل بكثير، ولكن قد لا يكون من الواضح كثيرًا كيفيّة تطبيق هذا في React. تشرح هذه الصفحة الخيارات التي وضعناها في خوارزمية المقارنة (diffing) بحيث تكون تحديثات المُكوّنات متوقعة وفي نفس الوقت سريعة كفاية لأجل التطبيقات عالية الأداء.
```
Comment (Fcmam5) [inline content/docs/reconciliation.md]: Please what's the most accurate translation of `declarative` is it `تعريفية` or `صريحة` ?

File: content/docs/reconciliation.md [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+عندما تستخدم React في نقطة زمنية محددة بإمكانك التفكير في التابع `render()` وكأنّه يُنشِئ شجرة من عناصر React،  وعند التحديث التالي للخاصيات `props` أو الحالة `state`  سيُعيد التابع `render()‎` شجرة مختلفة من عناصر React. تحتاج بعدها React لأن تعرف كيف ستُحدِّث واجهة المستخدم بكفاءة لُتطابِق آخر تحديث للشجرة. 
 
-There are some generic solutions to this algorithmic problem of generating the minimum number of operations to transform one tree into another. However, the [state of the art algorithms](https://grfia.dlsi.ua.es/ml/algorithms/references/editsurvey_bille.pdf) have a complexity in the order of O(n<sup>3</sup>) where n is the number of elements in the tree.
+هنالك بعض الحلول العامة لهذه المشكلة الحسابية لتوليد أقل عدد من العمليات المطلوبة للتحويل من شجرة إلى أخرى. على أية حال تمتلك [الخوارزميات النموذجيّة](https://grfia.dlsi.ua.es/ml/algorithms/references/editsurvey_bille.pdf) تعقيدًا من الترتيب O(n<sup>3</sup>) حيث n هو عدد العناصر الموجودة في الشجرة.
```
Comment (Fcmam5) [inline content/docs/reconciliation.md]: I translated `state of the art algorithms` to `الخوارزميات النموذجيّة` is it correct for this context?

File: content/docs/reconciliation.md (hunk 1 above)
Comment (iRayan7) [reply content/docs/reconciliation.md]: `صريحة` is a lot clearer 👍

File: content/docs/reconciliation.md (hunk 2 above)
Comment (iRayan7) [reply content/docs/reconciliation.md]: Spot on 👍

**Comment by Fcmam5** [comment]
> Amazing job mashallah tada
> I truly enjoy reading your translations heart

شرف عظيم لي أن أقرأ تعليقك أخي :)

**Pull request merged**